src/ode_control.cpp

```diff
@@ -42,11 +42,14 @@ double OdeControl::adjust_step_size(size_t dim, size_t ord,
     if (r < 0.2) {
       r = 0.2;
     }
-    step_size *= r;
-    last_step_size_shrank = true;
-    if (step_size < step_size_min) {
-      step_size = step_size_min;
-      util::stop("Step size became too small");
+    double new_step = step_size * r;
```

Comment:
Is `new step'`s only purpose to evaluate whether `step size` is too small. Is not returned later on except on line 51, so could possibly used a comment line to indicate that the `new_step` is not actually passed on as the `new_step` value.

Reply:
Thanks for checking, but actually it is potentially used in the next if statement, see 

```
    if(new_step < step_size) {
      step_size = new_step;
      last_step_size_shrank = true;
    }
 ```